### ISSUE TITLE
--silent-import fails inconsistently with partial stubs

### ISSUE THREAD

**Issue by sharmaeklavya2** [python/mypy#1549]
Consider the module `six.moves.configparser`. This module is not in typeshed, but parts of `six.moves` are in typeshed.

`mypy --silent-import -c 'import six.moves.configparser'` gives no error but `mypy --silent-import -c 'from six.moves import configparser'` gives an error:
```
file2.py:1: error: Module has no attribute 'configparser'
```

**Comment by gvanrossum** [comment]
I see no bug here. The `--silent-imports` stops warnings about modules it cannot find, like `six.moves.config` in the first example, but not about other attribute accesses. In the second example, it doesn't find a submodule `configparser` so it looks for an attribute by that name, on the package `six.moves` which itself was imported correctly. Since the package doesn't define that attribute, an error is reported.
